doc/tutorial/alignment_file/alignment_file_solution2.cpp

```diff
@@ -57,51 +57,58 @@ r003	2064	chr2	18	10	5M	*	0	0	TAGGC	*
 write_file_dummy_struct go{};
 
 //![solution]
-#include <seqan3/core/debug_stream.hpp>
-#include <seqan3/io/alignment_file/all.hpp>
-#include <seqan3/io/sequence_file/all.hpp>
+#include <string>
+#include <vector>
+
+#include <seqan3/alphabet/nucleotide/dna5.hpp>
 #include <seqan3/alphabet/gap/gap.hpp>
+#include <seqan3/core/debug_stream.hpp>
+#include <seqan3/io/alignment_file/input.hpp>
+#include <seqan3/io/record.hpp>
+#include <seqan3/io/sequence_file/input.hpp>
+#include <seqan3/range/container/concatenated_sequences.hpp>
 #include <seqan3/std/filesystem>
 #include <seqan3/std/ranges>
 
-using namespace seqan3;
-
-struct my_traits : public sequence_file_input_default_traits_dna
-{
-    //!\brief The container for sequences is now std::vector seqan3::concatenated_sequences.
-    template <typename _sequence_container>
-    using sequence_container_container = std::vector<_sequence_container>;
-};
-
 int main()
 {
     std::filesystem::path tmp_dir = std::filesystem::temp_directory_path(); // get the temp directory
 
     // read in reference information
-    sequence_file_input<my_traits> reference_file{tmp_dir/"reference.fasta"};
-    concatenated_sequences<std::string> ref_ids = get<field::ID>(reference_file);
-    std::vector<std::vector<dna5>> ref_seqs = get<field::SEQ>(reference_file);
+    seqan3::sequence_file_input reference_file{tmp_dir/"reference.fasta"};
+    seqan3::concatenated_sequences<std::string> ref_ids{};
+    std::vector<std::vector<seqan3::dna5>> ref_seqs{};
+
+    for (auto const & record : reference_file)
+    {
+        ref_ids.push_back(seqan3::get<seqan3::field::ID>(record));
+        ref_seqs.push_back(seqan3::get<seqan3::field::SEQ>(record));
+    }
 
-    alignment_file_input mapping_file{tmp_dir/"mapping.sam",
-                                      ref_ids,
-                                      ref_seqs,
-                                      fields<field::ID,field::REF_ID, field::MAPQ, field::ALIGNMENT>{}};
+    seqan3::alignment_file_input mapping_file{tmp_dir/"mapping.sam",
+                                              ref_ids,
+                                              ref_seqs,
+                                              seqan3::fields<seqan3::field::ID,
+                                                             seqan3::field::REF_ID,
+                                                             seqan3::field::MAPQ,
+                                                             seqan3::field::ALIGNMENT>{}};
 
-    auto mapq_filter = std::views::filter([] (auto & rec) { return get<field::MAPQ>(rec) >= 30; });
+    auto mapq_filter = std::views::filter([] (auto & rec) { return seqan3::get<seqan3::field::MAPQ>(rec) >= 30; });
```

Comment:
This `seqan3::get` works?

Reply:
yes, I see no error...

Reply:
Apparently, but I'm wondering why :)

Reply:
I guess because this `get` has a `seqan3::record` as argument...